### ISSUE TITLE
The 'Reader::setReadEmptyCells' method has no effect on Xlsx reader

### ISSUE THREAD

**Issue by phinor** [PHPOffice/PhpSpreadsheet#810]
This is:

```
- [x] a bug report
- [ ] a feature request
- [x] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
Empty cells will not be read from the spreadsheet and will not contribute to overall memory usage.

### What is the current behavior?
Empty cells are loaded, contributing to overall memory utilisation.

### What are the steps to reproduce?
In this example, I expect the two spreadsheets to have similar memory utilisation.

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$spreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();
$writer = \PhpOffice\PhpSpreadsheet\IOFactory::createWriter ($spreadsheet, "Xlsx");
$writer->save ("minimal.xlsx");

for ($row = 0; $row < 100; $row ++) {
    for ($column = 0; $column < 100; $column ++) {
        $spreadsheet->getActiveSheet ()->getCellByColumnAndRow ($column, $row)->setValue ("");
    }
}
$writer->save ("large.xlsx");

$reader = new \PhpOffice\PhpSpreadsheet\Reader\Xlsx();
$reader->setReadEmptyCells (false);

$memStart = memory_get_usage ();

$min = $reader->load ("minimal.xlsx");
$memMin = memory_get_usage ();
echo "Memory utilisation for minimal: " . ($memMin - $memStart) . "<br>";

$max = $reader->load ("large.xlsx");
$memMax = memory_get_usage ();
echo "Memory utilisation for large: " . ($memMax - $memMin) . "<br>";

for ($i = 0; $i < 5; $i ++) {
    $minAgain [] = $reader->load ("minimal.xlsx");
    $memMin = memory_get_usage ();
    echo "Memory utilisation for next minimal: " . ($memMin - $memMax) . "<br>";

    $maxAgain [] = $reader->load ("large.xlsx");
    $memMax = memory_get_usage ();
    echo "Memory utilisation for next large: " . ($memMax - $memMin) . "<br>";
}
```

### Which versions of PhpSpreadsheet and PHP are affected?
Currently using PHP 7.0.3; PhpSpreadsheet 1.5.2

**Comment by siarheipashkevich** [comment]
@phinor did you resolve your problem?

**Comment by phinor** [comment]
Yes. https://github.com/PHPOffice/PhpSpreadsheet/pull/835

**Comment by siarheipashkevich** [comment]
@phinor thanks

**Comment by synapt** [comment]
Was this possibly rolled back at some point or somehow not included into the composer builds?

I have now tested two Xlsx files, one generated in LibreOffice and another from MS Excel, neither are showing intended behavior from setReadEmptyCells(false) in either 1.8.1 or 1.9.0 (latest in composer).

Example code;
`
use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\IOFactory;
use PhpOffice\PhpSpreadsheet\Reader;

$reader = IOFactory::createReader('Xlsx');
$reader->setReadDataOnly(true);
$reader->setReadEmptyCells(false);
$spreadsheet = $reader->load($file)->getActiveSheet()->toArray();
var_dump($spreadsheet);
`

This does properly load the valid amount of rows, however it loads 1025 cells per row, populating all the empty ones as NULL.  Do one of you have an Xlsx that you were using for testing that I can confirm against?

Issue #468 may also be relevant to this but they never updated if they had resolved it.

**Comment by devnix** [comment]
The combination of `setReadDataOnly` and `setReadEmptyCells` is working for me with rows, but it keeps me reading a full last empty column!

**Comment by zkipart** [comment]
I'm afraid that it no longer works.
Version: 1.12.0
Xlsx attached

If I can help with more infos, I'm here. Thanks
[macheta import ibd (1).xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/4620144/macheta.import.ibd.1.xlsx)

**Comment by oleibman** [comment]
Initial problem seems to have been solved by PR #835, and nulls being added by `toArray` is explained by @cosminimum. Leaving issue closed, removing stale label.
